tests/contrib/hooks/test_sagemaker_hook.py

```diff
@@ -0,0 +1,246 @@
+            "VolumeSizeInGB": 50
+        },
+        "TrainingJobName": job_name,
+        "HyperParameters": {
+            "k": "10",
+            "feature_dim": "784",
+            "mini_batch_size": "500",
+            "force_dense": "True"
+        },
+        "StoppingCondition": {
+            "MaxRuntimeInSeconds": 60 * 60
+        },
+        "InputDataConfig": [
+            {
+                "ChannelName": "train",
+                "DataSource": {
+                    "S3DataSource": {
+                        "S3DataType": "S3Prefix",
+                        "S3Uri": data_location,
+                        "S3DataDistributionType": "FullyReplicated"
+                    }
+                },
+                "CompressionType": "None",
+                "RecordWrapperType": "None"
+            }
+        ]
+    }
+
+create_tuning_params = {'HyperParameterTuningJobName': job_name,
+                        'HyperParameterTuningJobConfig': {
+                            'Strategy': 'Bayesian',
+                            'HyperParameterTuningJobObjective': {
+                                'Type': 'Maximize',
+                                'MetricName': 'test_metric'
+                            },
+                            'ResourceLimits': {
+                                'MaxNumberOfTrainingJobs': 123,
+                                'MaxParallelTrainingJobs': 123
+                            },
+                            'ParameterRanges': {
+                                'IntegerParameterRanges': [
+                                    {
+                                        'Name': 'k',
+                                        'MinValue': '2',
+                                        'MaxValue': '10'
+                                    },
+                                ]
+                            }
+                        },
+                        'TrainingJobDefinition': {
+                            'StaticHyperParameters':
+                                create_training_params['HyperParameters'],
+                            'AlgorithmSpecification':
+                                create_training_params['AlgorithmSpecification'],
+                            'RoleArn': 'string',
+                            'InputDataConfig':
+                                create_training_params['InputDataConfig'],
+                            'OutputDataConfig':
+                                create_training_params['OutputDataConfig'],
+                            'ResourceConfig':
+                                create_training_params['ResourceConfig'],
+                            'StoppingCondition': dict(MaxRuntimeInSeconds=60 * 60)
+                        }
+                        }
+
+db_config = {"Re4sourceConfig": {
+    "InstanceCount": 3,
+    "InstanceType": "ml.c4.4xlarge",
+    "VolumeSizeInGB": 25
+}
+}
+
+
+class TestSageMakerHook(unittest.TestCase):
+
+    def setUp(self):
```

Comment:
I know setUp is same as emr hook test. But this doesn't look like python style. You can just do 'setup'.

Reply:
I think it is a keyword for Unittest, because I saw the python documentation does the same thing.